### PR TITLE
Add support for AMD Zen

### PR DESCRIPTION
**Pull request by AlGreyy**
Add support for AMD Zen platform

### CODE REVIEW

**Comment by vit9696** [comment]
Hi,

Please explain in detail the following questions:
- What you have (your setup, configuration, problems).
- What you are trying to do (with this pull request and in general).
- What you have done (with this pull request).

The code inside makes little sense, and I would rather not theorise.

Vit

File: Lilu/Sources/kern_devinfo.cpp [hunk 1]
```diff
@@ -273,9 +273,17 @@ void DeviceInfo::grabDevicesFromPciRoot(IORegistryEntry *pciRoot) {
 								v.video = pciobj;
 								v.vendor = pcivendor;
 							} else if (pcicode == WIOKit::ClassCode::HDADevice) {
+
```
Comment (07151129) [inline Lilu/Sources/kern_devinfo.cpp]: What does amdzen have to do with HDADevice?

**Comment by AlGreyy** [comment]
Please consider adding the changes as a commit in order to get Lilu to grabDevicesFromPciRoot correctly on AMD Zen motherboards. Those changes are required and in conjunction with this commit to AppleALC
https://github.com/AlGreyy/AppleALC/commit/f525345ccd28ae0c37db436f34cf7de94adf509a

**Comment by vit9696** [comment]
Thanks, I got it. I cleaned up the patch a little and merged it manually.

@07151129 I left a comment in https://github.com/acidanthera/Lilu/commit/5f147cf7cda622c6b7516a67a8c4c374cf4a4923#diff-52be878824893f6f1741487500b690a5R280 explaining the cause of this. New AMD platforms appear to have built-in HDEF audio sitting on PCI bridge instead of PCI root (like on Intel), so they would be falsely detected as a digital audio controller from the AMD GPU unless guarded by the vendor check.

**Comment by vit9696** [comment]
Hi,

Glad to hear it worked. The controller autodetection code should have been rewritten years ago, but stayed like this from like 2014 :D

As for the patch, cannot you patch out the Intel ID instead of breaking the GPUs? I can implement the logic you requested, but as a last effort at most.

Vit

File: Lilu/Sources/kern_devinfo.cpp (hunk 1 above)
Comment (yuqingdeng) [reply Lilu/Sources/kern_devinfo.cpp]: Nothing.  These are Raven and Zeppelin HD Audio Controllers.

**Pull request closed without merging**